umbral/umbral.py

```diff
@@ -22,6 +22,33 @@ def __init__(self, id_, key, x, u1, z1, z2):
         self.bn_sig1 = z1
         self.bn_sig2 = z2
 
+    @classmethod
+    def from_bytes(cls, data: bytes, curve):
+        """
+        Instantiate a KFrag object from the serialized data.
+        """
+        id = BigNum.from_bytes(data[0:32], curve)
+        key = BigNum.from_bytes(data[32:64], curve)
+        eph_ni = Point.from_bytes(data[64:97], curve)
+        commitment = Point.from_bytes(data[97:130], curve)
+        sig1 = BigNum.from_bytes(data[130:162], curve)
+        sig2 = BigNum.from_bytes(data[162:194], curve)
+
+        return KFrag(id, key, eph_ni, commitment, sig1, sig2)
+
+    def to_bytes(self):
+        """
+        Serialize the KFrag into a bytestring.
+        """
+        id = self.bn_id.to_bytes()
+        key = self.point_key.to_bytes()
+        eph_ni = self.point_eph_ni.to_bytes()
+        commitment = self.point_commitment.to_bytes()
+        sig1 = self.bn_sig1.to_bytes()
+        sig2 = self.bn_sig2.to_bytes()
+
+        return id + key + eph_ni + commitment + sig1 + sig2
```

Comment:
This method is a little rough I think - reading it, I have no idea why these particular attrs represent a canonical bytes repr.  At minimum, I think it's worth an additional comment.